### DISCUSSION TITLE
Nextcloud Calendar: "Strict Cookie has not been found in request"

### DISCUSSION THREAD

**Discussion by PublicBenemy** [derekantrican/GAS-ICS-Sync#457] [Q&A]
I have this public calendar URL:
https://ecodrom.next-cloud.org/remote.php/dav/public-calendars/QrzrE4wC47W5YmdB?export

It's freely accessible, there are no credentials necessary. When I open the URL in a browser, the ics is created instantly to download. When I use this URL in GAS-ICS-Sync, the return value is `{"error":"Strict Cookie has not been found in request"}`
When I click the URL with CTRL in the Apps Script editor, I receive the same result.

In Nextcloud log file, it says similarly "Request does not pass strict cookie check".

Any idea how to solve that?

**Comment by PublicBenemy** [comment]
Update: I was following the wrong path. GAS-ICS-Sync retrieves information, but it seems malformed. The "[WARNING] Microsoft is incorrectly formatting ics/ical at: ..." is thrown. Will analyse the returned ical now.

**Reply by jonas0b1011001** [reply]
This might still be due to the error in your first post if your nextcloud does not return an http error response code.

Unfortunately i don't know how the resolve that error.

**Reply by PublicBenemy** [reply]
Hi Jonas, thanks for passing by. On that calendar, the script actually receives plausible icsContent, but fails in parsing it somehow:
![grafik](https://github.com/user-attachments/assets/255eccb2-1fb7-46f1-b6d5-8018303cfed7)

I have two calendars on that Nextcloud. Interestingly enough, the other calendar gets a whole sync without any issues. It even doesn't throw the Microsoft warning thing. This only appears for the failing calendar, so I assume maybe there is just one event in an corrupt condition?

Edit 1: typo
Edit 2: the obfuscated URL in the screenshot is my personal calendar causing the trouble. The URL in the opening post is the one working flawlessly, it doesn't contain sensitive information.

**Reply by jonas0b1011001** [reply]
Maybe run it through an [ical validator](https://icalendar.org/validator.html).
